### DISCUSSION TITLE
Javascript - example of using the shared kotlin code consumed from javascript

### DISCUSSION THREAD

**Discussion by Phinneas** [reduxkotlin/redux-kotlin#120] [Q&A]
There are two uses: 
1) creating a kotlin js frontend and 
2) creating a js frontend that consumes the common kotlin code.

**Comment by mpetuska** [comment]
This is not kotlin-redux specific at all, but in short what you need is pack (and optionally publish) your kjs app as an npm module. From there you add it to js project just like any other npm module. [Here's](https://github.com/mpetuska/npm-publish/tree/master/sandbox/ts-consumer) one sample of mine how that can be achieved.
